src/database/database.ts

```diff
@@ -85,3 +110,122 @@ export class DatabaseService implements FirebaseServiceInterface {
     });
   }
 }
+
+const RULES_URL_PATH = '.settings/rules.json';
+
+/**
+ * A helper client for managing RTDB security rules.
+ */
+class DatabaseRulesClient {
+
+  private readonly httpClient: AuthorizedHttpClient;
+  private readonly dbUrl: string;
+
+  constructor(app: FirebaseApp, dbUrl: string) {
+    this.httpClient = new AuthorizedHttpClient(app);
+    this.dbUrl = dbUrl;
+  }
+
+  /**
+   * Gets the currently applied security rules as a string. The return value consists of
+   * the rules source including comments.
+   *
+   * @return {Promise<string>} A promise fulfilled with the rules as a raw string.
+   */
+  public getRules(): Promise<string> {
+    const req: HttpRequestConfig = {
+      method: 'GET',
+      url: this.getRulesUrl(),
+    };
+    return this.httpClient.send(req)
+      .then((resp) => {
+        return resp.text;
+      })
+      .catch((err) => {
+        throw this.handleError(err);
+      });
+  }
+
+  /**
+   * Gets the currently applied security rules as a parsed JSON object. Any comments in
+   * the original source are stripped away.
+   *
+   * @return {Promise<object>} A promise fulfilled with the parsed rules source.
+   */
+  public getRulesJSON(): Promise<object> {
+    const req: HttpRequestConfig = {
+      method: 'GET',
+      url: `${this.getRulesUrl()}?format=strict`,
+    };
+    return this.httpClient.send(req)
+      .then((resp) => {
+        return resp.data;
+      })
+      .catch((err) => {
+        throw this.handleError(err);
+      });
+  }
+
+  /**
+   * Sets the specified rules on the Firebase Database instance. If the rules source is
+   * specified as a string or a Buffer, it may include comments.
+   *
+   * @param {string|Buffer|object} source Source of the rules to apply. Must not be `null`
+   *  or empty.
+   * @return {Promise<void>} Resolves when the rules are set on the Database.
+   */
+  public setRules(source: string | Buffer | object): Promise<void> {
+    if (!validator.isNonEmptyString(source) &&
+      !validator.isBuffer(source)  &&
+      !validator.isNonNullObject(source)) {
+        throw new FirebaseDatabaseError({
```

Comment:
I am not sure about this pattern. It's tricky because although this looks like an async method, doing something like
```
db.setRules("foo").catch(err => {
```
will not actually catch this error (because it's thrown, not returned). Thoughts on this? I think it might be better to make this an `async` method or explicit return a `Promise.rejected` here

Reply:
We don't use async in this codebase yet.  Returning with a rejection instead.